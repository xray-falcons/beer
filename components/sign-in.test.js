import 'react-native';
import React from 'react';
import SignInView from "./sign-in";
import renderer from 'react-test-renderer';


describe('<SignInView />', () => {


    it('renders correctly', () => {

        const tree = renderer.create(
            <SignInView />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders the TextInput component', () => {
        const TextInput = require('TextInput');
        const tree = renderer
            .create(<TextInput />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});

