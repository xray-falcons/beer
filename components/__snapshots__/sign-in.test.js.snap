// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> renders correctly 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4290998025,
        4293835275,
      ],
      "endPoint": undefined,
      "locations": undefined,
      "startPoint": undefined,
    }
  }
  style={
    Object {
      "flex": 1,
      "height": "100%",
      "left": 0,
      "right": 0,
      "top": 0,
      "width": "100%",
    }
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={2}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</ViewManagerAdapter_ExpoLinearGradient>
`;

exports[`<Loading /> renders the ActivityIndicator component 1`] = `
<ActivityIndicator
  animating={true}
  color="#999999"
  hidesWhenStopped={true}
  size="large"
/>
`;

exports[`<SignInView /> renders correctly 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4290998025,
        4293835275,
      ],
      "endPoint": undefined,
      "locations": undefined,
      "startPoint": undefined,
    }
  }
  style={
    Object {
      "flex": 1,
      "height": "100%",
      "left": 0,
      "right": 0,
      "top": 0,
      "width": "100%",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "margin": 20,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Object {
          "borderRadius": 100,
          "height": 200,
          "marginBottom": 20,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 20,
          "width": 200,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomColor": "#F5FCFF",
          "borderBottomWidth": 1,
          "borderRadius": 30,
          "flexDirection": "row",
          "height": 35,
          "marginBottom": 2,
          "width": 200,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://png.icons8.com/message/ultraviolet/50/3498db",
          }
        }
        style={
          Object {
            "height": 15,
            "justifyContent": "center",
            "marginLeft": 15,
            "width": 15,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        keyboardType="email-address"
        onChangeText={[Function]}
        placeholder="Email"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "#FFFFFF",
            "flex": 1,
            "height": 40,
            "marginLeft": 16,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomColor": "#F5FCFF",
          "borderBottomWidth": 1,
          "borderRadius": 30,
          "flexDirection": "row",
          "height": 35,
          "marginBottom": 2,
          "width": 200,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://png.icons8.com/key-2/ultraviolet/50/3498db",
          }
        }
        style={
          Object {
            "height": 15,
            "justifyContent": "center",
            "marginLeft": 15,
            "width": 15,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Password"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "borderBottomColor": "#FFFFFF",
            "flex": 1,
            "height": 40,
            "marginLeft": 16,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 30,
            "flexDirection": "row",
            "height": 25,
            "justifyContent": "center",
            "marginBottom": 1,
            "width": 200,
          },
          Object {
            "backgroundColor": "#710000",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        Sign in
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 30,
            "flexDirection": "row",
            "height": 25,
            "justifyContent": "center",
            "marginBottom": 1,
            "width": 200,
          },
          Object {
            "backgroundColor": "#710000",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        Sign in with Facebook
      </Text>
    </View>
  </View>
</ViewManagerAdapter_ExpoLinearGradient>
`;

exports[`<SignInView /> renders the LinearGradient component 1`] = `
<TextInput
  allowFontScaling={true}
  rejectResponderTermination={true}
  underlineColorAndroid="transparent"
/>
`;

exports[`<SignInView /> renders the TextInput component 1`] = `
<TextInput
  allowFontScaling={true}
  rejectResponderTermination={true}
  underlineColorAndroid="transparent"
/>
`;
